src/libraries/System.Private.CoreLib/src/System/DateOnly.cs

```diff
@@ -0,0 +1,902 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System.Diagnostics;
+using System.Globalization;
+
+namespace System
+{
+    /// <summary>
+    /// represents dates with values ranging from January 1, 0001 Anno Domini (Common Era) through December 31, 9999 A.D. (C.E.) in the Gregorian calendar.
+    /// </summary>
+    public readonly struct DateOnly : IComparable, IComparable<DateOnly>, IEquatable<DateOnly>, IFormattable, ISpanFormattable
+    {
+        private readonly int _dayNumber;
+
+        // Maps to Jan 1st year 1
+        private const int MinDayNumber = 0;
+
+        // Maps to December 31 year 9999. The value calculated from "new DateTime(9999, 12, 31).Ticks / TimeSpan.TicksPerDay"
+        private const int MaxDayNumber = 3652058;
+
+        private static int DayNumberFromDateTime(DateTime dt) => (int)(dt.Ticks / TimeSpan.TicksPerDay);
+
+        private DateTime GetEquivalentDateTime() => new DateTime(_dayNumber * TimeSpan.TicksPerDay);
+
+        private DateOnly(int dayNumber)
+        {
+            Debug.Assert((uint)dayNumber <= 3652058);
+            _dayNumber = dayNumber;
+        }
+
+        /// <summary>
+        /// Return the instance of the DateOnly structure representing the minimal possible date can be created.
```

Comment:
```suggestion
        /// Gets the earliest possible date that can be created.
```

(I assume minimal means earliest.)